### ISSUE TITLE
TwbT v6.21/v6.20/v5.86/v5.85 will not load in 32-bit DFhack 0.43.05-r2

### ISSUE THREAD

**Issue by Thundercraft** [mifki/df-twbt#64]
I did a clean install of 32-bit DFhack 0.43.05-r2 onto **df_43_05_win32.zip** and then installed a recent release of the TwbT plugin. Specifically, I copied the **twbt.plug.dll** file into the /hack/plugins/ folder and copied the .png images into /data/art/ folder. However, I **_always_** get this error message in the DFhack window:

## > Can't load plugin twbt

I tried v6.21 first. When that failed, I tried v6.20, v5.86, and v5.85. And I made sure that I copied the plugin in the "0.43.05-r2" folder to match the DFhack 0.43.05-r2 that I'm using.

I **swear** that the problem is that recent TwbT releases will **only** work for 64-bit DFhack in a 64-bit Dwarf Fortress install. I've tried everything that I can think of. And I **did** get recent TwbT releases to work for 64-bit DFhack in a 64-bit Dwarf Fortress install, using the same install method I did for 32-bit versions.

If that's the case, then you're doing the DF community a disservice by not specifying in the release notes that these are only for 64-bit versions (or, at least, that they have yet to be tested in 32-bit versions). I wasted a lot of time scratching my head and trying to figure out what I was doing wrong...

**Comment by Thundercraft** [comment]
I forgot to mention: The other plugins included in the releases I mentioned (v6.21/v6.20/v5.86/v5.85) also seem incompatible with the 32-bit edition of DFhack 0.43.05-r2. Every time that I use them, I get similar error messages:

### > Can't load plugin mousequery
### > Can't load plugin automaterial
### > Can't load plugin resume

But DFhack **already comes** with these plugins. As such, I fail to see the purpose of including them in your releases. The version of these plugins that come in DFhack release are **guaranteed** to work with that release. But the version that comes with TwbT are not.

**Comment by pronvit** [comment]
You're completely right, I forgot to mention the readme to mention that it now supports 64bit only. And in general I need to update the readme to include other latest changes too. Documentation isn't my favourite part.

**Comment by Thundercraft** [comment]
At least that's cleared up. (In the back of my mind there was still a nagging voice questioning whether I was somehow mistaken.) Hopefully, this issue didn't confuse too many people.

Such things wouldn't even require documentation in the readme. All that's needed is one sentence, mentioned either in the release's description or posted on the forum. Heck, it may have been clear enough if you added "32-bit" to the filename or even just a "-32" suffix to the end.

**Comment by orffen** [comment]
Is this mainly a end-user support issue, or are there issues with actually compiling to 32-bit?

i.e. if I change the target platform to `/p:Platform=Win32` should I be able to compile the plugin?

**Comment by lethosor** [comment]
TWBT requires patches, which have only been found for 64-bit DF. Also, it probably lacks some of the code necessary to deal with both architectures, and it's possible that some 32-bit support code was removed during the transition to 64-bit, so it might not be as trivial as finding the patches.
